lib/kion/favorite.go

```diff
@@ -0,0 +1,119 @@
+package kion
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/kionsoftware/kion-cli/lib/structs"
+)
+
+////////////////////////////////////////////////////////////////////////////////
+//                                                                            //
+//  Favorites                                                                 //
+//                                                                            //
+////////////////////////////////////////////////////////////////////////////////
+
+// FavoritesResponse maps to the Kion API response.
+type FavoritesResponse struct {
+	Status    int                `json:"status"`
+	Favorites []structs.Favorite `json:"data"`
+}
+
+// GetAPIFavorites returns a list of a user's Favorites associated with a given Kion from the API
+func GetAPIFavorites(host string, token string) ([]structs.Favorite, int, error) {
+
+	url := fmt.Sprintf("%v/api/v3/user-cloud-access-role-alias", host)
+	query := map[string]string{}
+	var data interface{}
+	resp, statusCode, err := runQuery("GET", url, token, query, data)
+	if err != nil {
+		return nil, statusCode, err
+	}
+
+	// unmarshal response body
+	favResp := FavoritesResponse{}
+	jsonErr := json.Unmarshal(resp, &favResp)
+	if jsonErr != nil {
+		return nil, 0, err
+	}
+
+	var apiFavorites []structs.Favorite
+	for _, apiFav := range favResp.Favorites {
+		// normalize the access type to match what the CLI uses
+		if apiFav.AccessType == "console_access" {
+			apiFav.AccessType = "web"
+		} else if apiFav.AccessType == "short_term_key_access" {
+			apiFav.AccessType = "cli"
+		}
+
+		// if the upstream favorite has no name (alias), attempt to look up the account name
```

Comment:
This feels like a helpful thing to do. Are there any permission constraints that would make this call fail for read only or minimal perm users?